aptos-move/block-executor/src/executor.rs

```diff
@@ -483,6 +483,9 @@ where
                 scheduler.add_to_commit_queue(txn_idx);
             }
 
+            // An invariant check on the recorded outputs.
+            last_input_output.check_execution_status_during_commit(txn_idx)?;
```

Comment:
Question of whether we should also do it on line 441. @igor-aptos @zekun000

Reply:
if status is wrong because transaction was speculative, we shouldn't fail there?

only thing we can do there is check for DelayedFieldsCodeInvariantError? other errors are expected. but I think that one we might throw even earlier?

Reply:
so this just checks now that the status exists (if not, a lot of things might panic below, and here we can fallback to sequential, which should be strictly better).
once status is set, it checks it's not Aborted or Panic (those should short-circuit), and SpeculativeError (that shouldn't be committed). In these cases we also throw smt that will fallback, but given the signature of the method already allows it, it's easy and we lose nothing.

My question was, since the above logic (for delayed fields) also accesses the txn_last_input_output, maybe I can add it before there as well

Reply:
now that I think about it, I think it's fine to do, just extra invariant checks.

Reply:
we can have SpeculativeError, and then validate_commit_ready returning false and us re-executing transaction - and returning a different error

Reply:
oh yes, exactly! so maybe only that status exists or else fallback makes real sense there.

Reply:
I assume Abort can also be speculative (though don't know). 

so only code invariant there can/should end execution

Reply:
abort short circuits now.

Reply:
here everything is fine to short curcuit (I think). on line 441 we can only add code invariant to short circuit

Reply:
I meant abort short circuits during execution, so we should never see it here.

Reply:
we don't have speculative errors here since it's already rolling commit? any speculative should be fixed above with update_transaction_on_abort?

Reply:
yes, I meant adding an extra check before all that logic, but then speculative would be possible. I won't venture into that now, but we can do more limited checks there.